ngraph/core/src/op/util/logical_reduction.cpp

```diff
@@ -51,34 +51,27 @@ void op::util::LogicalReduction::set_reduction_axes(const AxisSet& reduction_axe
             ->output(0));
 }
 
-void op::util::LogicalReduction::validate_and_infer_types()
+PartialShape op::util::LogicalReduction::infer_reduction_output_shape(const bool keep_dims)
```

Comment:
Looks like this method do the same thing that `ArithmeticReduction::infer_reduction_output_shape` am I right? Can we reuse it?

Reply:
After we discussed I have introduced a new ReductionBase class with protected method to infer output shape. Please check this [commit](https://github.com/openvinotoolkit/openvino/pull/5609/commits/22f382e42f4e0692d0ed39194649eb3f2af8b1cb)